src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure/Commands/GCPerfSim/GCPerfSimFunctionalCommand.cs

```diff
@@ -0,0 +1,336 @@
+        private static readonly ISerializer _serializer = Common.Serializer;
+
+        public sealed class GCPerfSimFunctionalSettings : CommandSettings
+        {
+            [Description("Path to Configuration.")]
+            [CommandOption("-c|--configuration")]
+            public string? ConfigurationPath { get; init; }
+
+            [Description("Crank Server to target.")]
+            [CommandOption("-s|--server")]
+            public string? Server { get; init; }
+        }
+
+        internal static void SaveConfiguration(ConfigurationBase configuration, string outputPath, string fileName)
+        {
+            var serializedResult = _serializer.Serialize(configuration);
+            File.WriteAllText(Path.Combine(outputPath, fileName), serializedResult);
+        }
+
+        public override int Execute([NotNull] CommandContext context, [NotNull] GCPerfSimFunctionalSettings settings)
+        {
+            // I. Extract the configuration path.
+            string configurationPath = settings.ConfigurationPath;
+
+            // Parse out the yaml file -> Memory as a C# object make use of that.
+            // Precondition checks.
+            ConfigurationChecker.VerifyFile(configurationPath, $"{nameof(GCPerfSimFunctionalCommand)}");
+            GCPerfSimFunctionalConfiguration configuration = GCPerfSimFunctionalConfigurationParser.Parse(configurationPath, true);
+
+            // II. Create the test suite for gcperfsim functional tests.
+            string gcPerfSimOutputPath = Path.Combine(configuration.output_path, "GCPerfSim");
+            Core.Utilities.TryCreateDirectory(gcPerfSimOutputPath);
+
+            string suitePath = Path.Combine(configuration.output_path, "Suites");
+            string gcPerfSimSuitePath = Path.Combine(suitePath, "GCPerfSim_Functional");
+
+            Core.Utilities.TryCreateDirectory(gcPerfSimSuitePath);
+
+            // For each of the scenarios below:
+            // a. Add the coreruns. 
+            // b. Add the gcperfsim parameters that are pertinent to that run.
+            // c. Add any environment variables that will be related to that run.
+
+            // 1. Normal Server
+            CreateNormalServerSuite(gcPerfSimSuitePath, configuration);
+
+            // 2. Normal Workstation.
+            CreateNormalWorkstationSuite(gcPerfSimSuitePath, configuration);
+
+            // 3. LowMemoryContainer.
+            CreateLowMemoryContainerSuite(gcPerfSimSuitePath, configuration);
+
+            // 4. HighMemoryLoad.
+            CreateHighMemoryLoadSuite(gcPerfSimSuitePath, configuration);
+
+            // III. Execute all the functional tests.
+
+            string[] gcperfsimConfigurationFileNames = Directory.GetFiles(gcPerfSimSuitePath, "*.yaml");
+
+            Dictionary<string, GCPerfSimResults> yamlFileResultMap = new Dictionary<string, GCPerfSimResults>();
+
+            foreach (string gcperfsimConfigurationFileName in gcperfsimConfigurationFileNames)
+            {
```

Comment:
Put this in try-catch blocks - and continue if one run fails. Catch the exception and print out to the user.